spec/requests/dor_requests_spec.rb

```diff
@@ -54,6 +54,46 @@
       expect(response.body).to eq 'Object does not exist in the repository'
       expect(response.code).to eq '404'
     end
+
+    context 'when cocina is provided by caller' do
+      let(:cocina_model_json) { '{ "some": "json" }' }
+      let(:cocina_model_metadata) { instance_double(Dor::Services::Client::ObjectMetadata) }
+      let(:created_at) { '2022-02-27' }
+      let(:updated_at) { '2022-02-28' }
+
+      before do
+        allow(Dor::Services::Client::ObjectMetadata).to receive(:new).with(created_at: created_at, updated_at: updated_at).and_return(metadata)
+      end
+
+      it 'uses the provided cocina without hitting dor-services-app' do
+        allow(Cocina::Models).to receive(:build).with(JSON.parse(cocina_model_json)).and_return(cocina) # pretend our bogus test JSON is valid
+        post "/dor/reindex/#{druid}", params: { cocina_model_json: cocina_model_json, created_at: created_at, updated_at: updated_at }
+        expect(response.body).to eq "Successfully updated index for #{druid}"
+        expect(response.code).to eq '200'
+        expect(mock_solr_conn).to have_received(:add).with(mock_solr_doc, add_attributes: { commitWithin: 1000 })
+        expect(object_service).not_to have_received(:find_with_metadata)
+      end
+
+      it 'requires both the cocina json and the created_at/updated_at metadata' do
+        post "/dor/reindex/#{druid}", params: { cocina_model_json: cocina_model_json, updated_at: updated_at }
+        expect(response.code).to eq '200'
+        expect(mock_solr_conn).to have_received(:add).with(mock_solr_doc, add_attributes: { commitWithin: 1000 })
+        expect(object_service).to have_received(:find_with_metadata)
+      end
+
+      it 'provides the caller with a useful error if invalid cocina is provided' do
+        # Cocina::Models.build will be called with our bogus JSON, which will throw an error
+        allow(Honeybadger).to receive(:notify).and_call_original
```

Comment:
normally i wouldn't tack on `.and_call_original` here, but i wanted to make sure that the hack for addressing https://github.com/rails/rails/issues/43922 got exercised (see other review comment).

i just confirmed that if i leave this line as is, but take out the `request.session` in `DorController#reindex`, the test fails (if i just `allow` with no call to the original method, the error is papered over and the test passes with or without the hack).